### ISSUE TITLE
Codeclimate formatter generates invalid JSON syntax

### ISSUE THREAD

**Issue by pismy** [hadolint/hadolint#519]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

The Codeclimate formatter generates a **valid** report file (JSON syntax + schema).

### Actual behavior

Not the case.

Here is an example of an actual report:

```
{"location":{"path":"Dockerfile_ko","lines":{"begin":18,"end":18}},"severity":"major","check_name":"DL3027","categories":["Bug Risk"],"type":"issue","description":"Do not use apt as it is meant to be a end-user tool, use apt-get or apt-cache instead"}{"location":{"path":"Dockerfile_ko","lines":{"begin":48,"end":48}},"severity":"major","check_name":"SC1007","categories":["Bug Risk"],"type":"issue","description":"Remove space after = if trying to assign a value (for empty string, use var='' ... )."}{"location":{"path":"Dockerfile_ko","lines":{"begin":48,"end":48}},"severity":"major","check_name":"SC2154","categories":["Bug Risk"],"type":"issue","description":"node_verson is referenced but not assigned."}{"location":{"path":"Dockerfile_ko","lines":{"begin":48,"end":48}},"severity":"major","check_name":"DL3003","categories":["Bug Risk"],"type":"issue","description":"Use WORKDIR to switch to a directory"}
```

This is not even a valid JSON syntax.

(:warning: plus there are control characters not visible here)

### Steps to reproduce the behavior

Take any `Dockerfile` with issues, run `hadolint -f codeclimate Dockerfile`

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter v1.19.0-45-gef91156
```

**Comment by pismy** [comment]
After investigation:
* Code Climate Engine is expecting a **streamed output**, JSON formatted issues [begin separated by a null char](https://github.com/codeclimate/platform/blob/master/spec/analyzers/SPEC.md#output)
* other tools such as GitLab CI expect a non-streamed format, JSON compatible (each issue part of an array `[]`, separated by a coma)
